src/hotspot/cpu/riscv/vm_version_riscv.cpp

```diff
@@ -156,14 +156,35 @@ void VM_Version::initialize() {
     FLAG_SET_DEFAULT(UseSHA1Intrinsics, false);
   }
 
-  if (UseSHA256Intrinsics) {
-    warning("Intrinsics for SHA-224 and SHA-256 crypto hash functions not available on this CPU.");
-    FLAG_SET_DEFAULT(UseSHA256Intrinsics, false);
+  if (UseZvkn) {
+    if (!ext_V.enabled()) {
+      FLAG_SET_DEFAULT(UseZvkn, false);
+      warning("Cannot enable Zvkn on cpu without RVV support.");
+    }
+    if (FLAG_IS_DEFAULT(UseRVV)) {
+      FLAG_SET_DEFAULT(UseRVV, true);
+    }
+    if (UseRVV) {
+      if (FLAG_IS_DEFAULT(UseSHA256Intrinsics)) {
+        FLAG_SET_DEFAULT(UseSHA256Intrinsics, true);
```

Comment:
As the code comment in shared code indicates, there should be a dependency between these flags and `UseSHA`. But seems we are still lacking the necessary logic for that and `UseSHA` is always false [2]. We might need similar handling like our aarch64 counterpart [3].

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/globals.hpp#L342
[2] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/vm_version_riscv.cpp#L151
[3] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/vm_version_aarch64.cpp#L323-L376

Reply:
Fixed